bottlecap/src/config/flush_strategy.rs

```diff
@@ -26,28 +27,39 @@ impl<'de> Deserialize<'de> for FlushStrategy {
         } else {
             let mut split_value = value.as_str().split(',');
             // "periodically,60000"
-            match split_value.next() {
-                Some(first_value) if first_value.starts_with("periodically") => {
-                    let interval = split_value.next();
-                    // "60000"
-                    if let Some(interval) = interval {
-                        if let Ok(parsed_interval) = interval.parse() {
-                            return Ok(FlushStrategy::Periodically(PeriodicStrategy {
-                                interval: parsed_interval,
-                            }));
-                        }
-                        debug!("Invalid flush interval: {}, using default", interval);
-                        Ok(FlushStrategy::Default)
-                    } else {
-                        debug!("Invalid flush strategy: {}, using default", value);
-                        Ok(FlushStrategy::Default)
+            // "end,1000"
```

Comment:
this seems a great use cases for match-pattern

Reply:
I was thinking that at the beginning, but then I have to duplicate logic for interval parsing in both places + all the debugs when it fails, wdyt?

Reply:
Mh not sure,, the mapping can be done with
match
    end,periodic or periodic,end => do same stuff

right?

Reply:
Addressed it!